src/health_misinfo_shared/data_parsing.py

```diff
@@ -1,43 +1,9 @@
-import re
-import ast
-import json
-from typing import Any
+from typing import cast
 
-
-def tidy_response(response_text: str) -> str:
-    """Strip unnecessary head/tail of response string"""
-    # TODO: Maybe investigate this library instead: https://github.com/noamgat/lm-format-enforcer
-    response_text = response_text.strip(" `'\".")
-    if response_text.startswith("json"):
-        response_text = response_text[5:]
-    return response_text
-
-
-def parse_json_string(json_string: str) -> dict[str, Any] | None:
-    try:
-        return json.loads(json_string, strict=False)
-    except json.JSONDecodeError:
-        try:
-            return ast.literal_eval(json_string)
-        except Exception as e:
-            return None
+import json_repair
 
 
 def parse_model_json_output(model_output: str) -> list[dict[str, str]]:
-    original_output = model_output
-    model_output = tidy_response(model_output)
-    model_output = model_output.strip()
-    # model_output = re.sub(r"```", "", model_output.strip()).rstrip("\n").lstrip("json")
-    try:
-        if model_output.startswith("[") and model_output.endswith("]"):
-            return parse_json_string(model_output)
-
-        first_square_bracket_idx = model_output.find("[")
-        last_square_bracket_idx = model_output.rfind("]")
-        if first_square_bracket_idx >= 0 and last_square_bracket_idx >= 0:
-            return parse_json_string(
-                model_output[first_square_bracket_idx : last_square_bracket_idx + 1]
-            )
-    except Exception as e:
-        pass
-    raise Exception("Could not parse the string: ", model_output, original_output)
+    if not (parsed := json_repair.loads(model_output)):
+        raise Exception("Could not parse the string: ", model_output)
```

Comment:
You could be a little more specific and raise a ValueError (i.e. the value is the right type (str) but an inappropriate value)

Reply:
I’ve tried to change as little as possible from what was there before. Previously, this function raised an exception that looked like this:
```python
raise Exception("Could not parse the string: ", model_output, original_output)
```

So I’ve tried to be as consistent with that.

Reply:
Okay, I think you’re right – I’ll fix this.

I’ll change the exception raised to `ValueError` as you suggest, and I’ll update the corresponding test. I’ll leave the exception handling in the rest of the code as `except Exception` because it’s tricky to fix (the try/except blocks often include big chunks of code that could be raising other exceptions).